File: EventPlus.Backend/EventPlus.Application/Minis/Events/Models/EventModel.cs
using EventPlus.Application.Minis.Assignments.Models;
using EventPlus.Domain.Enums;

namespace EventPlus.Application.Minis.Events.Models;

public class EventModelBase
{
    public long Id { get; init; }

    public string Title { get; init; } = default!;
    public string? Description { get; init; }

    public long CreatorId { get; set; }
    public long CommandId { get; set; }

    public Priority Priority { get; init; }
    public DateTime Date { get; init; }
    
    public string Icon { get; set; }

    public DateTime Created { get; init; }
}

public class EventModelMini : EventModelBase
{
    public int AssignmentsCount { get; set; }
    public int CompletedAssignmentsCount { get; set; }
}

public class EventModel : EventModelBase
{
    public ICollection<AssignmentModel> Assignments { get; set; }
}

File: EventPlus.Frontend/api/event/types/eventModel.ts
import { AssignmentModel } from "~/api/assignment/types/assignmentModel";
import { Priority } from "~/api/types/Priority";

interface EventModelBase {
  id: number;

  title: string;
  description?: string;

  creatorId: number;
  commandId: number;

  priority: Priority;
  date: Date;

  icon: string;

  created: Date;
}

export interface EventModelMini extends EventModelBase {
  assignmentsCount: number;
  completedAssignmentsCount: number;
}

export interface EventModel extends EventModelBase {
  assignments: AssignmentModel[];
}
